modular_zubbers/code/modules/security/security_glock/gun.dm

```diff
@@ -41,70 +41,49 @@
 	. += span_notice("Spin reloads can be done on the move and need you only to hold your gun for their 2 second duration. ")
 	. += span_notice("If toggled on, spin reloads throw empty magazines in the direction you're facing, dealing damage dependant on their type. ")
 
-// /obj/item/gun/ballistic/automatic/pistol/sec_glock/security //This is what you give to Security Officers.
-// 	name = "\improper C-CK 9x25mm"
-// 	desc = "The Compact Criminal Killer, or C-CK9 for short, is a semi-automatic ballistic pistol meant for regulated station defense. These are normally issued with a special firing pin that only allows firing on code blue or higher."
-// 	icon = 'modular_zubbers/icons/obj/guns/sec_pistol.dmi'
-// 	icon_state = "black"
-// 	pin = /obj/item/firing_pin/alert_level/blue
-// 	spawn_magazine_type = /obj/item/ammo_box/magazine/m9mm/flathead
-// 	fire_delay = 4
-// 	can_suppress = FALSE
-// 	projectile_damage_multiplier = 0.65
-
-// /obj/item/gun/ballistic/automatic/pistol/sec_glock/security/rubber //This is what you give to cargo packages.
-// 	spawn_magazine_type = /obj/item/ammo_box/magazine/m9mm/rubber
-
 /obj/item/gun/ballistic/automatic/pistol/sec_glock/attack_self(mob/living/user)
 	var/obj/item/storage/our_belt = user.get_item_by_slot(ITEM_SLOT_BELT)
 	var/obj/item/ammo_box/magazine/m9mm/security/swapped_mag = locate(/obj/item/ammo_box/magazine/m9mm/security) in our_belt.contents
-	if(!magazine && swapped_mag)
-		if(swapped_mag)
-			if(insert_magazine(user, swapped_mag))
-				SpinAnimation(2, 1)
-				balloon_alert(user, "you spin your gun, loading in a mag!")
-			else
-				balloon_alert(user, "that magazine doesn't fit!")
+	if(!magazine && swapped_mag) //Empty gun, load in anyhow but without delay. No mag
+		if(insert_magazine(user, swapped_mag))
+			SpinAnimation(2, 1)
+			balloon_alert(user, "you spin your gun, loading in a mag!")
 		else
-			balloon_alert(user, "no spare magazines in your belt!")
-		return
+			balloon_alert(user, "that magazine doesn't fit!")
 	if(!chambered)
 		if(swapped_mag)
-			SpinAnimation(2, 1) //What a badass
-			if(do_after(user, 1 SECONDS, timed_action_flags = ( IGNORE_USER_LOC_CHANGE | IGNORE_TARGET_LOC_CHANGE ), target = src))
-				throw_eject_magazine(user, magejecting)
+			balloon_alert(user, "you spin your gun, loading in a mag!")
+			SpinAnimation(2, 1)
+			if(do_after(user, 1 SECONDS, timed_action_flags = (IGNORE_USER_LOC_CHANGE | IGNORE_TARGET_LOC_CHANGE), target = src))
+				throw_eject_magazine(user, magejecting, swapped_mag)
 		else
 			balloon_alert(user, "no spare magazines in your belt!")
-	if(bolt_type == BOLT_TYPE_NO_BOLT)
-		unload_ammo(user)
-		return
-	drop_bolt_silent(user)
+			return
+	drop_bolt(user)
 	if(recent_rack > world.time)
```

Comment:
oh yea use cooldown defines instead of manual world.time checks

Reply:
Unresolved because this still hasn't been done

Reply:
> Unresolved because this still hasn't been done

Discussed on discord with Artur, it's this way because the original proc is this way so changing just the subtype would be pointless and modifying base game procs is a bit much for the sake of one gun